src/Illuminate/Redis/RedisManager.php

```diff
@@ -84,11 +84,19 @@ public function connection($name = null)
         $name = $name ?: 'default';
 
         if (isset($this->connections[$name])) {
-            return $this->connections[$name];
+            // If the connection has had a fatal socket error, the \Redis object
+            // in this connection is no longer valid, and needs to be destroyed
+            // and recreated.
+            if (! $this->connections[$name]->shouldReconnect) {
+                return $this->connections[$name];
+            }
+            // Unset and destroy the Connection object to ensure it's cleaned up.
+            unset($this->connections[$name]);
```

Comment:
Note the unset is here to make sure that the isset always returns false, even if the connection below fails.